### PR TITLE
FUM-2863 - Deployment failure management

### PR DESCRIPTION
**Pull request by rbapst-tamedia**
# Timeout

With [wait_for_steady_state](https://registry.terraform.io/modules/terraform-aws-modules/ecs/aws/latest/submodules/service#input_wait_for_steady_state),  terraform apply will wait that the  new deployment is done before finishing. This PR allow to specify how long to wait before terraform times out and change the default (20 minutes) to 5 minutes. see [input_timeouts](https://registry.terraform.io/modules/terraform-aws-modules/ecs/aws/latest/submodules/service#input_timeouts) and [plugin_timeout](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/retries-and-customizable-timeouts) for details

# Circuit Breaker

Now if the deployment is not successful, terraform will fail but in the background AWS will still continue to apply the change. With the Circuit Breaker enabled (see [AWS doc on Circuit Breaker](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-circuit-breaker.html), AWS will stop to try to deploy service and roll-back to the previous deployment.

# Cosmetic
- Rename file to use "_" instead of "-" (align with DAI's best practice)
- Removed the "moved" blocks because it's here since the first release (v0.1.0) and nobody should still use an older version of this module (I checked and we don't)

# Notes

Still not 100% convienced by the "provider_timeouts" variable name.
- "timeouts" along is to vague
- "terraform_timeouts" ?
- "ecs_service_deployment_timeout" ?
What do you @tx-pts-dai/dai-team think ?

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -124,3 +124,23 @@ variable "stop_timeout" {
...
+variable "deployment_circuit_breaker" {
+  description = "(Optional) Configuration block for deployment circuit breaker"
+  type = object({
+    enable   = optional(bool, true)
+    rollback = optional(bool, true)
+  })
+  default = {}
+}
+
+variable "provider_timeouts" {
```
Comment (AlfGot) [inline variables.tf:138]: @rbapst-tamedia what about `service_timeouts`?

**Review by AlfGot — APPROVED** [review]
Cosmetic: the moved blocks can be removed.

**Comment by rbapst-tamedia** [comment]
> Cosmetic: the moved blocks can be removed.

@AlfGot which moved block ?

File: variables.tf (hunk 1 above)
Comment (rbapst-tamedia) [reply variables.tf:138]: yes because it's for the `aws_ecs_service`resource. Then why not `aws_ecs_service_timeout` ?

**Comment by AlfGot** [comment]
@rbapst-tamedia For ex:
`moved {
  from = aws_ecs_task_definition.ecs_service
  to   = aws_ecs_task_definition.this
}`

File: variables.tf (hunk 1 above)
Comment (AlfGot) [reply variables.tf:138]: For the circuit breaker we are not using the prefix `aws_ecs_`, therefore i would stick to just `service_timeouts`, but I let you decide. Maybe we can use the plurals (timeouts).

Comment (rbapst-tamedia) [reply variables.tf:138]: INFO: circuit_breaker is only for aws_ecs_service. Timeouts is a "reserved" variable for the provider. So we could have timeouts for aws_ec2_instances, aws_xyz...
But I agree, lets go with your proposition: `service_timeouts`.

**Comment by rbapst-tamedia** [comment]
> @rbapst-tamedia For ex: `moved { from = aws_ecs_task_definition.ecs_service to = aws_ecs_task_definition.this }`
Good point. I checked and the `moved` block is here since v0.1.0 of the module. I search in github and nobody is using a release earlier than v0.1.0. I search in the Ness state files and there is no reference to the old name. So good point to remove this.

**Pull request merged**